sidebar tests failing

When building the package for openSUSE, the test suite fails:
```
[   41s] =================================== FAILURES ===================================
[   41s] ____________________________ test_sidebars_captions ____________________________
[   41s] 
[   41s] sphinx_build_factory = <function sphinx_build_factory.<locals>._func at 0x7f22aee98280>
[   41s] file_regression = <pytest_regressions.file_regression.FileRegressionFixture object at 0x7f22aeebdf10>
[   41s] 
[   41s]     def test_sidebars_captions(sphinx_build_factory, file_regression):
[   41s]         sphinx_build = sphinx_build_factory("sidebars").build()
[   41s]     
[   41s]         subindex_html = sphinx_build.html_tree("section1/index.html")
[   41s]     
[   41s]         # Sidebar structure with caption
[   41s]         sidebar = subindex_html.select("nav#bd-docs-nav")[0]
[   41s] >       file_regression.check(sidebar.prettify(), extension=".html")
[   41s] E       AssertionError: FILES DIFFER:
[   41s] E       /tmp/pytest-of-abuild/pytest-0/test_sidebars_captions0/test_build/test_sidebars_captions.html
[   41s] E       /tmp/pytest-of-abuild/pytest-0/test_sidebars_captions0/test_build/test_sidebars_captions.obtained.html
[   41s] E       HTML DIFF: /tmp/pytest-of-abuild/pytest-0/test_sidebars_captions0/test_build/test_sidebars_captions.obtained.diff.html
[   41s] E       --- 
[   41s] E       +++ 
[   41s] E       @@ -1,6 +1,6 @@
[   41s] E        <nav aria-label="Main navigation" class="bd-links" id="bd-docs-nav">
[   41s] E         <div class="bd-toc-item active">
[   41s] E       -  <p class="caption">
[   41s] E       +  <p class="caption" role="heading">
[   41s] E           <span class="caption-text">
[   41s] E            Section 1
[   41s] E           </span>
[   41s] 
[   41s] tests/test_build.py:205: AssertionError
[   41s] __________________________ test_sidebars_nested_page ___________________________
[   41s] 
[   41s] sphinx_build_factory = <function sphinx_build_factory.<locals>._func at 0x7f22af1e1d30>
[   41s] file_regression = <pytest_regressions.file_regression.FileRegressionFixture object at 0x7f22af27f1f0>
[   41s] 
[   41s]     def test_sidebars_nested_page(sphinx_build_factory, file_regression):
[   41s]         sphinx_build = sphinx_build_factory("sidebars").build()
[   41s]     
[   41s]         subindex_html = sphinx_build.html_tree("section1/subsection1/page1.html")
[   41s]     
[   41s]         # For nested (uncollapsed) page, the label included `checked=""`
[   41s]         sidebar = subindex_html.select("nav#bd-docs-nav")[0]
[   41s] >       file_regression.check(sidebar.prettify(), extension=".html")
[   41s] E       AssertionError: FILES DIFFER:
[   41s] E       /tmp/pytest-of-abuild/pytest-0/test_sidebars_nested_page0/test_build/test_sidebars_nested_page.html
[   41s] E       /tmp/pytest-of-abuild/pytest-0/test_sidebars_nested_page0/test_build/test_sidebars_nested_page.obtained.html
[   41s] E       HTML DIFF: /tmp/pytest-of-abuild/pytest-0/test_sidebars_nested_page0/test_build/test_sidebars_nested_page.obtained.diff.html
[   41s] E       --- 
[   41s] E       +++ 
[   41s] E       @@ -1,6 +1,6 @@
[   41s] E        <nav aria-label="Main navigation" class="bd-links" id="bd-docs-nav">
[   41s] E         <div class="bd-toc-item active">
[   41s] E       -  <p class="caption">
[   41s] E       +  <p class="caption" role="heading">
[   41s] E           <span class="caption-text">
[   41s] E            Section 1
[   41s] E           </span>
[   41s] 
[   41s] tests/test_build.py:215: AssertionError
[   41s] _____________________________ test_sidebars_single _____________________________
[   41s] 
[   41s] sphinx_build_factory = <function sphinx_build_factory.<locals>._func at 0x7f22aef583a0>
[   41s] file_regression = <pytest_regressions.file_regression.FileRegressionFixture object at 0x7f22aedd7dc0>
[   41s] 
[   41s]     def test_sidebars_single(sphinx_build_factory, file_regression):
[   41s]         confoverrides = {"templates_path": ["_templates_single_sidebar"]}
[   41s]         sphinx_build = sphinx_build_factory("sidebars", confoverrides=confoverrides).build()
[   41s]     
[   41s]         index_html = sphinx_build.html_tree("index.html")
[   41s]     
[   41s]         # No navbar included
[   41s]         assert not index_html.select("nav#navbar-main")
[   41s]         assert not index_html.select(".navbar-nav")
[   41s]     
[   41s]         # Sidebar structure
[   41s]         sidebar = index_html.select("nav#bd-docs-nav")[0]
[   41s] >       file_regression.check(sidebar.prettify(), extension=".html")
[   41s] E       AssertionError: FILES DIFFER:
[   41s] E       /tmp/pytest-of-abuild/pytest-0/test_sidebars_single0/test_build/test_sidebars_single.html
[   41s] E       /tmp/pytest-of-abuild/pytest-0/test_sidebars_single0/test_build/test_sidebars_single.obtained.html
[   41s] E       HTML DIFF: /tmp/pytest-of-abuild/pytest-0/test_sidebars_single0/test_build/test_sidebars_single.obtained.diff.html
[   41s] E       --- 
[   41s] E       +++ 
[   41s] E       @@ -1,7 +1,7 @@
[   41s] E        <nav aria-label="Main navigation" class="bd-links" id="bd-docs-nav">
[   41s] E         <div class="bd-toc-item active">
[   41s] E          <!-- Specify a startdepth of 0 instead of default of 1 -->
[   41s] E       -  <p class="caption">
[   41s] E       +  <p class="caption" role="heading">
[   41s] E           <span class="caption-text">
[   41s] E            Caption 1
[   41s] E           </span>
[   41s] E       @@ -13,7 +13,7 @@
[   41s] E            </a>
[   41s] E           </li>
[   41s] E          </ul>
[   41s] E       -  <p class="caption">
[   41s] E       +  <p class="caption" role="heading">
[   41s] E           <span class="caption-text">
[   41s] E            Caption 2
[   41s] E           </span>
[   41s] 
[   41s] tests/test_build.py:230: AssertionError
[   41s] _____________________________ test_sidebars_level2 _____________________________
[   41s] 
[   41s] sphinx_build_factory = <function sphinx_build_factory.<locals>._func at 0x7f22af0a33a0>
[   41s] file_regression = <pytest_regressions.file_regression.FileRegressionFixture object at 0x7f22afba8cd0>
[   41s] 
[   41s]     def test_sidebars_level2(sphinx_build_factory, file_regression):
[   41s]         confoverrides = {"templates_path": ["_templates_sidebar_level2"]}
[   41s]         sphinx_build = sphinx_build_factory("sidebars", confoverrides=confoverrides).build()
[   41s]     
[   41s]         subindex_html = sphinx_build.html_tree("section1/subsection1/index.html")
[   41s]     
[   41s]         # Sidebar structure
[   41s]         sidebar = subindex_html.select("nav#bd-docs-nav")[0]
[   41s] >       file_regression.check(sidebar.prettify(), extension=".html")
[   41s] E       AssertionError: FILES DIFFER:
[   41s] E       /tmp/pytest-of-abuild/pytest-0/test_sidebars_level20/test_build/test_sidebars_level2.html
[   41s] E       /tmp/pytest-of-abuild/pytest-0/test_sidebars_level20/test_build/test_sidebars_level2.obtained.html
[   41s] E       HTML DIFF: /tmp/pytest-of-abuild/pytest-0/test_sidebars_level20/test_build/test_sidebars_level2.obtained.diff.html
[   41s] E       --- 
[   41s] E       +++ 
[   41s] E       @@ -1,7 +1,7 @@
[   41s] E        <nav aria-label="Main navigation" class="bd-links" id="bd-docs-nav">
[   41s] E         <div class="bd-toc-item active">
[   41s] E          <!-- Use deeper level for sidebar -->
[   41s] E       -  <p class="caption">
[   41s] E       +  <p class="caption" role="heading">
[   41s] E           <span class="caption-text">
[   41s] E            Subsection 1.1
[   41s] E           </span>
[   41s] 
[   41s] tests/test_build.py:241: AssertionError
[   41s] =========================== short test summary info ============================
[   41s] FAILED tests/test_build.py::test_sidebars_captions - AssertionError: FILES DI...
[   41s] FAILED tests/test_build.py::test_sidebars_nested_page - AssertionError: FILES...
[   41s] FAILED tests/test_build.py::test_sidebars_single - AssertionError: FILES DIFFER:
[   41s] FAILED tests/test_build.py::test_sidebars_level2 - AssertionError: FILES DIFFER:
[   41s] ========================= 4 failed, 27 passed in 7.80s =========================
[   41s] error: Bad exit status from /var/tmp/rpm-tmp.aSatb5 (%check)
```

Using Sphinx 4.1.1 and any other packages mentioned in  [the full build log](https://github.com/pydata/pydata-sphinx-theme/files/6896749/_log.txt).


https://github.com/pydata/pydata-sphinx-theme/issues/389

--- a/tests/test_build/test_sidebars_captions.html
+++ b/tests/test_build/test_sidebars_captions.html
@@ -1,6 +1,6 @@
 <nav aria-label="Main navigation" class="bd-links" id="bd-docs-nav">
  <div class="bd-toc-item active">
-  <p class="caption">
+  <p class="caption" role="heading">
    <span class="caption-text">
     Section 1
    </span>

--- a/tests/test_build/test_sidebars_level2.html
+++ b/tests/test_build/test_sidebars_level2.html
@@ -1,7 +1,7 @@
 <nav aria-label="Main navigation" class="bd-links" id="bd-docs-nav">
  <div class="bd-toc-item active">
   <!-- Use deeper level for sidebar -->
-  <p class="caption">
+  <p class="caption" role="heading">
    <span class="caption-text">
     Subsection 1.1
    </span>

--- a/tests/test_build/test_sidebars_nested_page.html
+++ b/tests/test_build/test_sidebars_nested_page.html
@@ -1,6 +1,6 @@
 <nav aria-label="Main navigation" class="bd-links" id="bd-docs-nav">
  <div class="bd-toc-item active">
-  <p class="caption">
+  <p class="caption" role="heading">
    <span class="caption-text">
     Section 1
    </span>

--- a/tests/test_build/test_sidebars_single.html
+++ b/tests/test_build/test_sidebars_single.html
@@ -1,7 +1,7 @@
 <nav aria-label="Main navigation" class="bd-links" id="bd-docs-nav">
  <div class="bd-toc-item active">
   <!-- Specify a startdepth of 0 instead of default of 1 -->
-  <p class="caption">
+  <p class="caption" role="heading">
    <span class="caption-text">
     Caption 1
    </span>
@@ -13,7 +13,7 @@
     </a>
    </li>
   </ul>
-  <p class="caption">
+  <p class="caption" role="heading">
    <span class="caption-text">
     Caption 2
    </span>

